src/python/tests/test_statistics_history.py

```diff
@@ -0,0 +1,160 @@
+
+    def teardown_method(self):
+        """Set up each test with fresh data, and empty _KEYS dictionary"""
+        _KEYS.clear()
+        os.system("psql < `locate foglamp_ddl.sql | grep 'FogLAMP/src/sql'` > /dev/null 2>&1")
+        os.system("psql < `locate foglamp_init_data.sql | grep 'FogLAMP/src/sql'` > /dev/null 2>&1")
+
+    async def test_get_key_list(self):
+        """
+        Test that_get_key_list function works properly by comparing to the
+        initial data
+        :assert:
+             The sorted list of keys returned is equal to the keys currently retreived
+        """
+        result = _list_stats_keys()
+        assert sorted(result) == sorted(_KEYS)
+
+    async def test_insert_into_stats_history(self):
+        """
+        Test that _insert_into_stats_history updates the value per key
+        :assert:
+            Assert statistics_history.value == rand_value for a given key
+        """
+        for key in _KEYS:
+            stmt = sa.select([_STATS_HISTORY_TABLE.c.value]).where(
+                _STATS_HISTORY_TABLE.c.key == key)
+            rand_value = random.randint(1, 10)
+            _insert_into_stats_history(key=key, value=rand_value,
+                                       history_ts=str(datetime.datetime.now()))
+            try:
+                async with aiopg.sa.create_engine(_CONNECTION_STRING) as engine:
+                    async with engine.acquire() as conn:
+                        async for result in conn.execute(stmt):
+                            assert result[0] == rand_value
+            except Exception:
+                raise
+
+    async def test_update_previous_value(self):
+        """
+        Test the update of statistics.previous_value
+        :assert:
+            Assert  statistics.previous_value  gets updated with the set value
+        """
+        for key in _KEYS:
+            stmt = sa.select([_STATS_TABLE.c.value, _STATS_TABLE.c.previous_value]).where(
+                _STATS_TABLE.c.key == key)
+            rand_value = random.randint(1, 10)
+            _update_previous_value(key=key, value=rand_value)
+            try:
+                async with aiopg.sa.create_engine(_CONNECTION_STRING) as engine:
+                    async with engine.acquire() as conn:
+                        async for result in conn.execute(stmt):
+                            assert result[0] == 0
+                            assert result[1] == rand_value
+            except Exception:
+                raise
+
+    async def test_select_from_statistics(self):
+        """
+        Test that _select_from_statistics retrieves value and previous value from statistics table
+        :assert:
+            Assert that both value and previous value == 0
+        """
+        for key in _KEYS:
+            result = _select_from_statistics(key=key)
+            assert result[0][0] == 0
+            assert result[0][0] == result[0][1]
+
+    async def test_stats_history_main(self):
```

Comment:
I'm not clear what this test is doing?! we can make it simpler with non random values and may be break into 2-3 tests?